pkg/tls/renewer.go

```diff
@@ -116,46 +161,59 @@ func (c *CertRenewer) getSecret(name string) (*corev1.Secret, error) {
 	}
 }
 
-func (c *CertRenewer) getCASecret() (*corev1.Secret, error) {
-	return c.getSecret(GenerateRootCASecretName())
+func (c *CertRenewer) decodeSecret(name string) (*corev1.Secret, *rsa.PrivateKey, []*x509.Certificate, error) {
+	secret, err := c.getSecret(name)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	var certBytes, keyBytes []byte
+	if secret != nil {
+		keyBytes = secret.Data[corev1.TLSPrivateKeyKey]
+		certBytes = secret.Data[corev1.TLSCertKey]
+		if len(certBytes) == 0 {
+			certBytes = secret.Data[RootCAKey]
+		}
+	}
+	var key *rsa.PrivateKey
+	if keyBytes != nil {
+		usedkey, err := pemToPrivateKey(keyBytes)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		key = usedkey
+	}
+	return secret, key, pemToCertificates(certBytes), nil
 }
 
-func (c *CertRenewer) getTLSSecret() (*corev1.Secret, error) {
-	return c.getSecret(GenerateTLSPairSecretName())
+func (c *CertRenewer) decodeCASecret() (*corev1.Secret, *rsa.PrivateKey, []*x509.Certificate, error) {
```

Comment:
Normally we will only have one CA cert, and there will be 2 CAs during renewal. Correct?